### DISCUSSION TITLE
Facing issue when running a test on lambda

### DISCUSSION THREAD

**Discussion by srikrishnaj2001** [artilleryio/artillery#2145] [Artillery]
Here is the error thats displayed in console 

Test run id: tp5c9_kewezptnggekkfhmbt6cczbxn3zm4_xmgm
NOTE: AWS Lambda support is experimental. Not all Artillery features work yet.
For details please see https://docs.art/aws-lambda

λ Creating AWS Lambda function...
- Bundling test data
  - outskill-webinar.yml
- Installing dependencies
- Creating zip package
Preparing AWS environment...
 - Lambda function: artilleryio-tp5c9_kewezptnggekkfhmbt6cczbxn3zm4_xmgm
 - Region: ap-south-1
 - AWS account: ************* ( My AWS Account Number)
[2815d55a-c357-41c0-a83d-024103a7e16f]: A Lambda function has exited with an error. Reason: ArtilleryError
{
  stdout: '',
  stderr: 'node:internal/modules/cjs/loader:1031\n' +
    '  throw err;\n' +
    '  ^\n' +
    '\n' +
    "Error: Cannot find module '@oclif/core'\n" +
    'Require stack:\n' +
    '- /var/task/node_modules/artillery/bin/run\n' +
    '    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1028:15)\n' +
    '    at Function.Module._load (node:internal/modules/cjs/loader:873:27)\n' +
    '    at Module.require (node:internal/modules/cjs/loader:1100:19)\n' +
    '    at require (node:internal/modules/cjs/helpers:119:18)\n' +
    '    at Object.<anonymous> (/var/task/node_modules/artillery/bin/run:8:16)\n' +
    '    at Module._compile (node:internal/modules/cjs/loader:1198:14)\n' +
    '    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1252:10)\n' +
    '    at Module.load (node:internal/modules/cjs/loader:1076:32)\n' +
    '    at Function.Module._load (node:internal/modules/cjs/loader:911:12)\n' +
    '    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12) {\n' +
    "  code: 'MODULE_NOT_FOUND',\n" +
    "  requireStack: [ '/var/task/node_modules/artillery/bin/run' ]\n" +
    '}\n'
}

Getting the error even for the basic script thats in Documentation.

**Comment by bernardobridge** [comment]
Hi @srikrishnaj2001,

Thanks for reporting this. We've had a couple of users report this, but so far haven't been able to replicate it. 

Can you please share the following information:

1. Are you or your company using an npm caching proxy or a private registry of any kind (e.g. Artifactory, Verdaccio, Cloudsmith etc)?
2. Can you please share the command you ran and your folder structure?
3. Can you please share the artillery version (`artillery --version`) and OS version you used?
4. Can you please share the `node` and `npm` version you're using?
5. If you run `npm config ls -l | grep -E install-strategy\|legacy-bundling\|global-style`, can you please tell me the values of `install-strategy`, `legacy-bundling` and `global-style`? (this checks the default values of those options, which may affect how modules are bundled).

Thank you!

**Reply by srikrishnaj2001** [reply]
Here are the details :

Yes we use NPM Artifactory which expires every 12 hours

Folder Structure - /Desktop/loadtest/tests/loadtest.yml

VERSION INFO:

Artillery: 2.0.0-37
Node.js:   v20.7.0
OS:        darwin

npm version - 10.1.0

global-style = false 
install-strategy = "hoisted" 
legacy-bundling = false

**Comment by bernardobridge** [comment]
Hey @srikrishnaj2001 ,

Thank you so much for the information! That's very helpful 🙇‍♂️.

This might be related with the use of Artifactory with Lambda, as everything else looks in order. We'll try to replicate this on our side, but bear with us, as it may take a bit to get Artifactory setup to test it. 

In the mean time, you should be able to use `Fargate` without a problem with `Artifactory`, and it should be just as simple to setup, if you want an alternative to Lambda.
